relayer/application_relayer.go

```diff
@@ -102,30 +106,84 @@ func newApplicationRelayer(
 	checkpointManager := checkpoint.NewCheckpointManager(logger, db, sub, relayerID, startingHeight)
 	checkpointManager.Run()
 
-	ar := applicationRelayer{
+	ar := ApplicationRelayer{
 		logger:            logger,
 		metrics:           metrics,
 		network:           network,
 		messageCreator:    messageCreator,
-		responseChan:      responseChan,
 		sourceBlockchain:  sourceBlockchain,
 		relayerID:         relayerID,
 		signingSubnetID:   signingSubnet,
 		warpQuorum:        quorum,
 		checkpointManager: checkpointManager,
+		currentRequestID:  rand.Uint32(), // TODONOW: pass via ctor
+		lock:              &sync.RWMutex{},
 	}
 
 	return &ar, nil
 }
 
-func (r *applicationRelayer) relayMessage(
-	unsignedMessage *avalancheWarp.UnsignedMessage,
+// Process [msgs] at height [height] by relaying each message to the destination chain.
+// Checkpoints the height with the checkpoint manager when all messages are relayed.
```

Comment:
Could you add to this comment explaining that an `ApplicationRelayer` instance expects `ProcessHeight` to be called for every block greater than or equal to its start block height?